test/base-tlsutility.cpp

```diff
@@ -85,71 +46,230 @@ BOOST_AUTO_TEST_CASE(sha1)
 	}
 }
 
-BOOST_AUTO_TEST_CASE(iscauptodate_ok)
+static String GetOpenSSLError()
 {
-	auto key (GenKeypair());
-
-	BOOST_CHECK(IsCaUptodate(MakeCert("Icinga CA", key, "Icinga CA", key, [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
-		BOOST_REQUIRE(X509_gmtime_adj(notBefore, 0));
-		BOOST_REQUIRE(X509_gmtime_adj(notAfter, LEAF_VALID_FOR + 60 * 60));
-	}).get()));
+	std::array<char, 256> errBuf;
+	ERR_error_string_n(ERR_get_error(), errBuf.data(), errBuf.size());
+	return {errBuf.data()};
 }
 
-BOOST_AUTO_TEST_CASE(iscauptodate_expiring)
+static std::shared_ptr<EVP_PKEY> GetEVP_PKEY(const String& keyfile)
 {
-	auto key (GenKeypair());
+	BIO *cakeybio = BIO_new_file(keyfile.CStr(), "r");
+	BOOST_REQUIRE_MESSAGE(cakeybio, "BIO_new_file() for private key from'" << keyfile << "': " << GetOpenSSLError());
+
+	RSA* rsa = PEM_read_bio_RSAPrivateKey(cakeybio, nullptr, nullptr, nullptr);
+	BOOST_REQUIRE_MESSAGE(rsa, "PEM read bio RSA key from private key file '" << keyfile << "': " << GetOpenSSLError());
+
+	BIO_free(cakeybio);
+	BOOST_CHECK_EQUAL(1, RSA_check_key(rsa)); // 1 == valid, 0 == invalid
 
-	BOOST_CHECK(!IsCaUptodate(MakeCert("Icinga CA", key, "Icinga CA", key, [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
-		BOOST_REQUIRE(X509_gmtime_adj(notBefore, 0));
-		BOOST_REQUIRE(X509_gmtime_adj(notAfter, LEAF_VALID_FOR - 60 * 60));
-	}).get()));
+	EVP_PKEY *pkey = EVP_PKEY_new();
+	EVP_PKEY_assign_RSA(pkey, rsa);
+	BOOST_CHECK_EQUAL(EVP_PKEY_RSA, EVP_PKEY_id(pkey));
+
+	return {pkey, EVP_PKEY_free};
 }
 
-BOOST_AUTO_TEST_CASE(iscertuptodate_ok)
+/**
+ * Creates a new X509 certificate signed by the Icinga CA based on an existing CA certificate.
+ *
+ * It uses the public key and subject name from the existing CA certificate to create a new certificate.
+ *
+ * @param ca The existing CA certificate.
+ * @param validFor The validity period in seconds.
+ * @param validFrom The start time offset in seconds.
+ * @param caCert Whether to create a CA certificate (true) or a leaf certificate (false).
+ * @returns The new X509 certificate or an empty shared_ptr on error.
+ */
+auto NewCertFromExisting(const std::shared_ptr<X509>& ca, long validFrom, long validFor, bool caCert = false)
```

Comment:
Should probably be declared `static` as well (just like the other helper functions).